mambaSharedFramework/HLSParser.swift

```diff
@@ -282,6 +282,11 @@ fileprivate final class HLSParseWorker: NSObject, HLSRapidParserCallback {
     }
     
     func startParse() {
+        // special case if data is empty return an empty array of tags
```

Comment:
this felt like the best place to put this check since it was the last place before moving over to the obj-c interface for the c parser. I'm open to moving if there's a better location

Reply:
I suspect this would still crash with a "logically" empty HLS manifest, like `" "` (A file with one space) or `"\n"`.

Can you test for that? Ideally the fix would be "never crash, even when we detect no tags".

Reply:
added a unit test for `" "` and `"\n"`, both produce "valid" playlists without crashing:
```
# Generated by Mamba(1.0.5) Copyright (c) 2017 Comcast Corporation
 
```

Reply:
OK. I should have been more upfront with my objection, which is that this is more of a bandaid than a fix.

Somewhere there's code that is assuming that an array is always at least 1 element long. This fix is preventing that code from executing, but there might be other ways of making it happen, or someone else might write code in the future to execute the bad code.

I'd rather we found the real culprit. I suspect it's in the C Rapid Parser code, which looks daunting, but is simpler than it appears. Possibly in the `parseHLS` function?